File: backend/app/utils/enums.py
from enum import Enum


class PerfilUsuario(str, Enum):
    """Perfis de usuario conforme docs/01-visao-geral.md"""
    ADMINISTRADOR = "ADMINISTRADOR"
    COORDENADOR = "COORDENADOR"
    GESTOR_POLO = "GESTOR_POLO"
    APOIO_COORDENADOR = "APOIO_COORDENADOR"


class FonteFinanciamento(str, Enum):
    """Fontes de financiamento dos projetos"""
    EMBRAPII = "EMBRAPII"
    EMPRESA = "EMPRESA"
    SEBRAE = "SEBRAE"
    IFPB = "IFPB"


class TipoSolicitacao(str, Enum):
    """Tipos de solicitacao de RH"""
    IMPLANTACAO = "IMPLANTACAO"
    ALTERACAO = "ALTERACAO"
    PAGAMENTO = "PAGAMENTO"


class StatusSolicitacao(str, Enum):
    """Status da solicitacao"""
    EM_EDICAO = "EM_EDICAO"
    SUBMETIDA = "SUBMETIDA"
    APROVADA = "APROVADA"
    REJEITADA = "REJEITADA"


class StatusVersaoRH(str, Enum):
    """Status da versao de RH do projeto"""
    PROPOSTA = "PROPOSTA"
    VIGENTE = "VIGENTE"
    HISTORICO = "HISTORICO"


class StatusTransferencia(str, Enum):
    """Status da transferencia de pesquisador"""
    PENDENTE = "PENDENTE"
    ACEITA = "ACEITA"
    RECUSADA = "RECUSADA"


class StatusProjeto(str, Enum):
    """Status do projeto"""
    ATIVO = "ATIVO"
    FINALIZADO = "FINALIZADO"
    SUSPENSO = "SUSPENSO"


class CategoriaBolsa(str, Enum):
    """Categorias de bolsa conforme Resolucao 11/2022"""
    PESQUISADOR_MASTER = "PESQUISADOR_MASTER"
    PESQUISADOR_SENIOR = "PESQUISADOR_SENIOR"
    PESQUISADOR_PLENO = "PESQUISADOR_PLENO"
    PESQUISADOR_JUNIOR = "PESQUISADOR_JUNIOR"
    PROFISSIONAL_SENIOR = "PROFISSIONAL_SENIOR"
    PROFISSIONAL_PLENO = "PROFISSIONAL_PLENO"
    PROFISSIONAL_JUNIOR = "PROFISSIONAL_JUNIOR"
    PROFISSIONAL_INICIANTE = "PROFISSIONAL_INICIANTE"
    ESTUDANTE_SUPERIOR_AVANCADO = "ESTUDANTE_SUPERIOR_AVANCADO"
    ESTUDANTE_SUPERIOR_INTERMEDIARIO = "ESTUDANTE_SUPERIOR_INTERMEDIARIO"
    ESTUDANTE_SUPERIOR_INICIANTE = "ESTUDANTE_SUPERIOR_INICIANTE"
    ESTUDANTE_MEDIO = "ESTUDANTE_MEDIO"


class TipoParametroRegra(str, Enum):
    """Tipos de parametros/regras"""
    VALOR_BOLSA = "VALOR_BOLSA"
    LIMITE_CARGA_HORARIA = "LIMITE_CARGA_HORARIA"

File: frontend/src/types/enums.ts
export enum FonteFinanciamento {
  EMBRAPII = 'EMBRAPII',
  EMPRESA = 'EMPRESA',
  SEBRAE = 'SEBRAE',
  IFPB = 'IFPB',
}

export enum TipoSolicitacao {
  IMPLANTACAO = 'IMPLANTACAO',
  ALTERACAO = 'ALTERACAO',
  PAGAMENTO = 'PAGAMENTO',
}

export enum TipoJustificativaSolicitacao {
  IMPLANTACAO = 'IMPLANTACAO',
  ALTERACAO = 'ALTERACAO',
  REJEICAO = 'REJEICAO',
}

export enum TipoDocumentoProjeto {
  ACORDO_PARCEIRA = 'ACORDO_PARCEIRA',
  PLANO_TRABALHO = 'PLANO_TRABALHO',
  DIARIO_OFICIAL = 'DIARIO_OFICIAL',
  DOCUMENTO_COMPLEMENTAR = 'DOCUMENTO_COMPLEMENTAR',
}

export enum StatusSolicitacao {
  EM_EDICAO = 'EM_EDICAO',
  SUBMETIDA = 'SUBMETIDA',
  APROVADA = 'APROVADA',
  REJEITADA = 'REJEITADA',
}

export enum StatusVersaoRH {
  PROPOSTA = 'PROPOSTA',
  VIGENTE = 'VIGENTE',
  HISTORICO = 'HISTORICO',
}

export enum StatusProjeto {
  ATIVO = 'ATIVO',
  FINALIZADO = 'FINALIZADO',
  SUSPENSO = 'SUSPENSO',
}

export enum CategoriaBolsa {
  PESQUISADOR_MASTER = 'PESQUISADOR_MASTER',
  PESQUISADOR_SENIOR = 'PESQUISADOR_SENIOR',
  PESQUISADOR_PLENO = 'PESQUISADOR_PLENO',
  PESQUISADOR_JUNIOR = 'PESQUISADOR_JUNIOR',
  PROFISSIONAL_SENIOR = 'PROFISSIONAL_SENIOR',
  PROFISSIONAL_PLENO = 'PROFISSIONAL_PLENO',
  PROFISSIONAL_JUNIOR = 'PROFISSIONAL_JUNIOR',
  PROFISSIONAL_INICIANTE = 'PROFISSIONAL_INICIANTE',
  ESTUDANTE_SUPERIOR_AVANCADO = 'ESTUDANTE_SUPERIOR_AVANCADO',
  ESTUDANTE_SUPERIOR_INTERMEDIARIO = 'ESTUDANTE_SUPERIOR_INTERMEDIARIO',
  ESTUDANTE_SUPERIOR_INICIANTE = 'ESTUDANTE_SUPERIOR_INICIANTE',
  ESTUDANTE_MEDIO = 'ESTUDANTE_MEDIO',
}

export const FONTE_LABELS: Record<FonteFinanciamento, string> = {
  [FonteFinanciamento.EMBRAPII]: 'EMBRAPII',
  [FonteFinanciamento.EMPRESA]: 'Empresa',
  [FonteFinanciamento.SEBRAE]: 'SEBRAE',
  [FonteFinanciamento.IFPB]: 'IFPB',
}

export const FONTES_RH_OPERACIONAIS: FonteFinanciamento[] = [
  FonteFinanciamento.EMPRESA,
  FonteFinanciamento.EMBRAPII,
  FonteFinanciamento.SEBRAE,
]

export const STATUS_SOLICITACAO_LABELS: Record<StatusSolicitacao, string> = {
  [StatusSolicitacao.EM_EDICAO]: 'Em Edição',
  [StatusSolicitacao.SUBMETIDA]: 'Submetida',
  [StatusSolicitacao.APROVADA]: 'Aprovada',
  [StatusSolicitacao.REJEITADA]: 'Rejeitada',
}

export const STATUS_PROJETO_LABELS: Record<StatusProjeto, string> = {
  [StatusProjeto.ATIVO]: 'Ativo',
  [StatusProjeto.FINALIZADO]: 'Finalizado',
  [StatusProjeto.SUSPENSO]: 'Suspenso',
}

export const TIPO_SOLICITACAO_LABELS: Record<TipoSolicitacao, string> = {
  [TipoSolicitacao.IMPLANTACAO]: 'Implantação',
  [TipoSolicitacao.ALTERACAO]: 'Alteração',
  [TipoSolicitacao.PAGAMENTO]: 'Pagamento',
}

export const TIPO_DOCUMENTO_PROJETO_LABELS: Record<TipoDocumentoProjeto, string> = {
  [TipoDocumentoProjeto.ACORDO_PARCEIRA]: 'Acordo de Parceria',
  [TipoDocumentoProjeto.PLANO_TRABALHO]: 'Plano de Trabalho',
  [TipoDocumentoProjeto.DIARIO_OFICIAL]: 'Diário Oficial',
  [TipoDocumentoProjeto.DOCUMENTO_COMPLEMENTAR]: 'Documento Complementar',
}
